openmc/deplete/results.py

```diff
@@ -0,0 +1,496 @@
+        float
+            The atoms for stage, mat, nuc
+
+        """
+        stage, mat, nuc = pos
+        if isinstance(mat, str):
+            mat = self.mat_to_ind[mat]
+        if isinstance(nuc, str):
+            nuc = self.nuc_to_ind[nuc]
+
+        return self.data[stage, mat, nuc]
+
+    def __setitem__(self, pos, val):
+        """Sets an item from results.
+
+        Parameters
+        ----------
+        pos : tuple
+            A three-length tuple containing a stage index, mat index and a nuc
+            index.  All can be integers or slices.  The second two can be
+            strings corresponding to their respective dictionary.
+
+        val : float
+            The value to set data to.
+
+        """
+        stage, mat, nuc = pos
+        if isinstance(mat, str):
+            mat = self.mat_to_ind[mat]
+        if isinstance(nuc, str):
+            nuc = self.nuc_to_ind[nuc]
+
+        self.data[stage, mat, nuc] = val
+
+    @property
+    def n_mat(self):
+        return len(self.mat_to_ind)
+
+    @property
+    def n_nuc(self):
+        return len(self.nuc_to_ind)
+
+    @property
+    def n_hdf5_mats(self):
+        return len(self.mat_to_hdf5_ind)
+
+    @property
+    def n_stages(self):
+        return self.data.shape[0]
+
+    def allocate(self, volume, nuc_list, burn_list, full_burn_list, stages):
+        """Allocates memory of Results.
+
+        Parameters
+        ----------
+        volume : dict of str float
+            Volumes corresponding to materials in full_burn_dict
+        nuc_list : list of str
+            A list of all nuclide names. Used for sorting the simulation.
+        burn_list : list of int
+            A list of all mat IDs to be burned.  Used for sorting the simulation.
+        full_burn_list : list of str
+            List of all burnable material IDs
+        stages : int
+            Number of stages in simulation.
+
+        """
+        self.volume = copy.deepcopy(volume)
+        self.nuc_to_ind = {nuc: i for i, nuc in enumerate(nuc_list)}
+        self.mat_to_ind = {mat: i for i, mat in enumerate(burn_list)}
+        self.mat_to_hdf5_ind = {mat: i for i, mat in enumerate(full_burn_list)}
+
+        # Create storage array
+        self.data = np.zeros((stages, self.n_mat, self.n_nuc))
+
+    def distribute(self, local_materials, ranges):
+        """Create a new object containing data for distributed materials
+
+        Parameters
+        ----------
+        local_materials : iterable of str
+            Materials for this process
+        ranges : iterable of int
+            Slice-like object indicating indicies of ``local_materials``
+            in the material dimension of :attr:`data` and each element
+            in :attr:`rates`
+
+        Returns
+        -------
+        Results
+            New results object
+        """
+        new = Results()
+        new.volume = {lm: self.volume[lm] for lm in local_materials}
+        new.mat_to_ind = dict(zip(
```

Comment:
so this is just enumerating, and flipping the indexes. I think this would be much clearer:
`{mat: idx for (idx, mat) in enumerate(local_materials)]}`

This shows that you are 1. enumerating the local_materials 2. constructing a dictionary out of it 
in one line.